src/v/crypto/random.cc

```diff
@@ -37,4 +37,32 @@ bytes generate_random(size_t len, use_private_rng private_rng) {
     generate_random(ret, private_rng);
     return ret;
 }
+
+template<>
+secure_private_rng::result_type secure_private_rng::operator()() {
+    result_type x = 0;
+    if (
+      1
+      != RAND_priv_bytes_ex(
+        nullptr, reinterpret_cast<unsigned char*>(&x), sizeof(x), 0)) {
+        throw internal::ossl_error(
+          "Failed to generate private random for secure_private_rng");
+    }
+
+    return x;
```

Comment:
Unless you actually want a different error message, the "correct" way to do this is:
```suggestion
    std::array<bytes::value_type, sizeof(result_type)> x;
    generate_random({x.begin(), sizeof(result_type)}, use_private_rng::yes);
    return std::bit_cast<result_type>(x);
```

Reply:
fair point, thanks